plugins/repository-s3/README.md

```diff
@@ -0,0 +1,21 @@
+# repository-s3
```

Comment:
Thank you @dblock for taking effort on adding this documentation. Care to add same for other similar plugins e.g. repository-gcs, repository-azure, repository-hdfs ?

Reply:
I really don't know anything about the other repository plugins other than at a high level, would prefer if the next person to work on them dug up the details and documented them.